shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -593,7 +596,148 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], newDefs: Bo
+        err((
+          msg"quote syntax is not supported yet." -> S(l0) :: Nil))
+        R(Var("<error>"))
+      case ((v, r), L(acc)) => L(IfLet(false, v, r, acc))
+    }
+  }
+
+  private def ifStmt(genQuote: Bool, l0: Loc)(implicit et: ExpectThen, fe: FoundErr, l: Line, qenv: Set[Str], quoted: IsQuoted): IfBody \/ Term =
+    exprOrIf(0)(et = true, fe = fe, l = implicitly, qenv = qenv, quoted = quoted) match {
+      case L(body) =>
+        val els = yeetSpaces match {
+          case (KEYWORD("else"), _) :: _ =>
+            consume
+            S(expr(0))
+          case (NEWLINE, _) :: (KEYWORD("else"), _) :: _ =>
+            consume
+            consume
+            S(expr(0))
+          case (br @ BRACKETS(Indent, (KEYWORD("else"), _) :: toks), _) :: _ =>
+            consume
+            val nested = rec(toks, S(br.innerLoc), br.describe)
+            S(nested.concludeWith(_.expr(0)))
+          case _ => N
+        }
+        if (!genQuote) R(If(body, els))
+        else body match {
+          case IfThen(cond, body) =>
+            R(Quoted(If(IfThen(Unquoted(cond), Unquoted(body)), els.map(els => Unquoted(els)))))
+          case _ =>
+            err((
+              msg"quote syntax is not supported yet." -> S(l0) :: Nil))
+            R(Var("<error>"))
+        }
+      case R(e) if !genQuote =>
+        yeetSpaces match {
+          case (br @ BRACKETS(Indent, (KEYWORD("then"), _) :: toks), _) :: _ =>
+            consume
+            val nested = rec(toks, S(br.innerLoc), br.describe)
+            val thn = nested.expr(0)
+            val els = nested.yeetSpaces match {
+              case (KEYWORD("else"), _) :: _ =>
+                nested.consume
+                S(nested.concludeWith(_.expr(0)))
+              case (NEWLINE, _) :: (KEYWORD("else"), _) :: _ =>
+                nested.consume
+                nested.consume
+                // S(thn, S(nested.concludeWith(_.expr(0))))
+                S(nested.concludeWith(_.expr(0)))
+              case _ =>
+                nested.concludeWith(_.nil)
+                // S(thn, N)
+                N
+            }
+            R(If(IfThen(e, thn), els))
+          case _cur =>
+            val (found, loc) = _cur match {
+              case (tk, l1) :: _ => (msg"${e.describe} followed by ${tk.describe}",
+                S(e.toLoc.foldRight(l1)(_ ++ _)))
+              case Nil => (msg"${e.describe}", e.toLoc)
+            }
+            err((msg"Expected 'then'/'else' clause after 'if'; found $found instead" -> loc ::
+              msg"Note: 'if' expression starts here:" -> S(l0) :: Nil))
+            R(If(IfThen(e, errExpr), N))
+        }
+      case R(_) =>
+        err((
+          msg"quote syntax is not supported yet." -> S(l0) :: Nil))
+        R(Var("<error>"))
+    }
+
+  private def bracket(genQuote: Bool, br: BRACKETS, bk: BracketKind, toks: Ls[Stroken -> Loc], loc: Loc)(
+    implicit et: ExpectThen, fe: FoundErr, l: Line, qenv: Set[Str], quoted: IsQuoted
```

Comment:
Let's not wrap variables in `code"."` anymore. We want `x` to have type `Var[T, C] <: Code[T, C]` in `x `=> ... x ...`, and we could simply also allow `code"x => ${ ... x ... }"`